### PR TITLE
[FI] Areas rework part 1 of many

### PR DESCRIPTION
**Pull request by salleq**
- Added conjugations for areas

- simplified home_assistant_HassTurnOn 

- made some other changes to common and fixtures

### CODE REVIEW

**Comment by Arkkimaagi** [comment]
I've been thinking this approach. I think it's quite problematic and we need to consider it more.

You've added support for two special rooms, ullakko and eteinen, but what if the people have "kura-eteinen" or something similar? I mean, we really can not anticipate all the rooms and names people give to their areas.

**Comment by v1k70rk4** [comment]
Hi, since we’re part of the same language family, I ran a test, and as I see, it works in Finnish too :) 

In Hungarian, there are some words where the root changes, so I added "filler words" after the area. For example: room, space, and it works.

So, for cases where the word cannot be conjugated, you can still use these. In your case, it might look something like this: Sytytä ullakko huoneen lamppu. Of course, words that can be conjugated without altering the root can still be supported.

This is precisely why I suggested creating a guide in everyone’s native language to explain usage. But until we can modify word roots, I’ve been solving it in Hungarian in this way.

Consider whether this works in all special cases.

**Comment by Arkkimaagi** [comment]
Yep, that's the option salleq is changing with this pr away from.

And the words fo not behave nicely, for eteinen, the extra word does not work as well.

It's also kinda difficult to remember what filler words to add.

**Comment by salleq** [comment]
This PR _adds_ the option for conjugated room names, I'm _not_ taking away the possibility to say "laita huoneen olohuone valo päälle" ("put the light on in the room living room"). You can see that I have added some tests, but not removed any.

We can't possibly take into account every room name there is (like "kura-eteinen") and what people will use, but the most common ones we can + some rooms that aren't conjugated nicely (ullakko and eteinen for example). 

I don't see how this could be better implemented unless we get Arkkimaagi's lemmatization idea go through or something similar.

It's more for us to maintain (having both conjugations and area {area} ) but I can't agree on talking to the voice assistant using filler words, it's not natural.

**Review by Arkkimaagi — CHANGES_REQUESTED** [review]
All in all, I love how you've taken care of this project on the Finnish translations. Since we've been discussing in discord and you asked for my opinion, I've checked the code. We can discuss further.

File: sentences/fi/_common.yaml [hunk 1]
```diff
@@ -345,8 +346,12 @@ expansion_rules:
   # laitteille: "[äly]laitteille"
   # laitteitta: "[äly]laitteitta"
 
+  # Special treatment for certain rooms
+  eteinen_pääte: "(eteinen|eteisessä|eteisestä|eteiseen|eteiselle|eteiseltä|eteisen)"
```
[the remark this answers is not in the document]
Comment (salleq) [reply sentences/fi/_common.yaml:350]: Just tested this - Had a light in a room called "Ullakko", and asked HA to "Sammuta ullakon valo". So it works.

File: sentences/fi/_common.yaml [hunk 2]
```diff
@@ -358,6 +363,7 @@ expansion_rules:
   alueelta: "(alueelta|paikalta|huoneelta|tilalta)" # ablatiivi = miltä, keneltä
   alueelle: "(alueelle|paikalle|huoneelle|tilalle)" # allatiivi = mille, kenelle
   # alueetta: "(alueetta|paikatta|huoneetta|tilatta)" # abessiivi = ilman mitä, ketä
+  alue_pääte: "({area}[e][ssa|ssä|sta|stä|än|in|an|on|ön|seen|ltä|llä|lla|lta|lle|n])"
```
[the remark this answers is not in the document]
Comment (salleq) [reply sentences/fi/_common.yaml:367]: as discussed, this probably won't matter even though the variants are multiplied. But - can be tested if necessary.

File: sentences/fi/_common.yaml [hunk 3]
```diff
@@ -345,8 +346,12 @@ expansion_rules:
   # laitteille: "[äly]laitteille"
   # laitteitta: "[äly]laitteitta"
 
+  # Special treatment for certain rooms
+  eteinen_pääte: "(eteinen|eteisessä|eteisestä|eteiseen|eteiselle|eteiseltä|eteisen)"
+  ullakko_pääte: "(ullakko|ullakolta|ullakolla|ullakolle|ullakon)"
+
   # Area prefix to avoid congjugating areas if possible
-  alue: "(alue|paikka|huone|tila)" # nominatiivi = mikä, kuka
+  alue: "(alue|paikka|huone|tila|<alueen>|<alueessa>|<alueesta>|<alueeseen>|<alueella>|<alueelta>|<alueelle>)" # nominatiivi = mikä, kuka
```
[the remark this answers is not in the document]
Comment (salleq) [reply sentences/fi/_common.yaml]: "alue" is now the base and "alue_taivutus" includes all the conjugations.

File: sentences/fi/fan_HassTurnOff.yaml [hunk 4]
```diff
@@ -3,70 +3,53 @@ intents:
...
-          - "<kytke> <päältä> <tuuletin> (<alueessa>|<alueesta>|<alueeseen>|<alueella>|<alueelta>|<alueelle>) <huone>"
-          - "<kytke> <päältä> (<alueen>|<alueessa>|<alueesta>|<alueeseen>|<alueella>|<alueelta>|<alueelle>) <huone> <tuuletin>"
-          - "<kytke> <tuuletin> <päältä> (<alueessa>|<alueesta>|<alueeseen>|<alueella>|<alueelta>|<alueelle>) <huone>"
-          - "<kytke> <tuuletin> (<alueessa>|<alueesta>|<alueeseen>|<alueella>|<alueelta>|<alueelle>) <huone> <päältä>"
-          - "<kytke> (<alueessa>|<alueesta>|<alueeseen>|<alueella>|<alueelta>|<alueelle>) <huone> <päältä> <tuuletin>"
-          - "<kytke> (<alueen>|<alueessa>|<alueesta>|<alueeseen>|<alueella>|<alueelta>|<alueelle>) <huone> <tuuletin> <päältä>"
-          - "<pysäytä> <tuuletin> (<alueessa>|<alueesta>|<alueeseen>|<alueella>|<alueelta>|<alueelle>) <huone>"
-          - "<pysäytä> (<alueen>|<alueessa>|<alueesta>|<alueeseen>|<alueella>|<alueelta>|<alueelle>) <huone> <tuuletin>"
+          # Yksittäinen tuuletin
+          - "<pysäytä> (<tuuletin>;(<huone>|<alue> <huone>|<alue_pääte>))"
```
[the remark this answers is not in the document]
Comment (salleq) [reply sentences/fi/fan_HassTurnOff.yaml]: discussed quite a bit about all these issues today in discord. if something doesn't seem how we agreed, please let me know.

**Comment by salleq** [comment]
There's more to be done obviously but I think I should freeze this PR with these changes unless there's something really out of place. 

I'll clean out the commented out stuff that's to be deleted, before merging this.

**Review by Arkkimaagi — APPROVED** [review]

**Pull request merged**